tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file test_signal_generator.cpp
+ * @author Luciana Moreno (lumoreno.borges@gmail.com)
+ * @brief Unit test for the signal generator hyro package
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+#include <hyro/test/ComponentTester.h>
+#include <gtest/gtest.h>
+
+/* Macro definition to configure the components */
+static void
+__assert_and_run_state_machine(hyro::StateMachine& sm,
+                               const hyro::ComponentConfiguration& conf,
+                               const hyro::ConnectionConfiguration& endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+#define ASSERT_AND_RUN_STATE_MACHINE(sm, conf, endpoint) \
+  __assert_and_run_state_machine(sm, conf, endpoint)
+
+namespace hyro
+{
+namespace test
+{
+
+/**
+ * @brief Construct a new TEST object for SignalGeneratorComponent
+ * 
+ */
+
+TEST (TestSignalGenerator, SignalGeneratorComponentCheck)
+{
+
+  auto configuration = ComponentConfiguration("{" 
+  "outputs: {"
+    "signals: { protocol: 'api'},"
+    "fix_dynamic: {protocol: 'api'}"
+    "}" 
+    "}"
+  );
+
+  
+  StateMachine signal_generator_sm(std::make_shared<SignalGeneratorComponent>("/signal_generator"_uri));
+  
+  ASSERT_AND_RUN_STATE_MACHINE(signal_generator_sm, configuration, ConnectionConfiguration());
+
+  auto fake_input_signals = std::make_shared<FakeInput<Signal>>("fake_input_signals"_uri, "api", "/signal_generator/signals");
+  
+  ASSERT_TRUE(fake_input_signals->connect());
+
+  signal_generator_sm.update();
+
+  auto fake_input_value = std::shared_ptr<const Signal>();
+
+  ReceiveStatus ret = fake_input_signals->receive(fake_input_value, 500ms);
+  
+  ASSERT_EQ(ReceiveStatus::RECEIVE_OK, ret);
+  EXPECT_EQ(fake_input_value->value, 0.0);
```

Comment:
Did you run the tests? This one is failing.

```
/home/joao/dev/workspace/src/examples/signal_generator/tests/test_signal_generator.cpp:72: Failure
Expected equality of these values:
  fake_input_value->value
    Which is: 3
  0.0
    Which is: 0

```
```
[  FAILED  ] TestSignalGenerator.SignalGeneratorComponentCheck (1413 ms)
[----------] 1 test from TestSignalGenerator (1413 ms total)

[----------] 1 test from DigitalConverterTest
[ RUN      ] DigitalConverterTest.DynamicPropertyCheck
[       OK ] DigitalConverterTest.DynamicPropertyCheck (24 ms)
[----------] 1 test from DigitalConverterTest (24 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 2 test cases ran. (1437 ms total)
[  PASSED  ] 1 test.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestSignalGenerator.SignalGeneratorComponentCheck

 1 FAILED TEST

```

Reply:
@joaobrittoneto  Fixed!